src/main/java/com/twilio/http/Request.java

```diff
@@ -175,9 +176,28 @@ private String buildURL() {
                 host = joinIgnoreNull(".", product, targetEdge, targetRegion, domain);
             }
 
-            return new URI(parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
-                    parsedUrl.getPath(), parsedUrl.getQuery(), parsedUrl.getRef()).toURL().toString();
-        } catch (final MalformedURLException | URISyntaxException e) {
+            String urlPort = null;
+            if (parsedUrl.getPort() != -1) {
+                urlPort = ":" + parsedUrl.getPort();
+            }
+            String protocol = parsedUrl.getProtocol() + "://";
+            String[] pathPieces = parsedUrl.getPath().split("/", 0);
+            pathPieces[pathPieces.length-1] = URLEncoder.encode(pathPieces[pathPieces.length-1], "UTF-8");
+            String encodedPath = Joiner.on("/").join(pathPieces);
+            String query = null;
+            if (parsedUrl.getQuery() != null) {
+                query = "?" + parsedUrl.getQuery();
+            }
```

Comment:
Simplify these for readability: `String query = parsedUrl.getQuery() != null ? "?" + parsedUrl.getQuery() : null;`